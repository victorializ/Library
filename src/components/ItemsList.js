import React  from 'react';

import { Card } from 'semantic-ui-react';

function ItemsList({data, Component, ...rest}) {
  return (
    <Card.Group>
      {
        data.map(element => 
          <Component 
            key={element.id} 
            item={element} 
            {...rest}
          />) 
      }
    </Card.Group>
  );
}

export { ItemsList };
