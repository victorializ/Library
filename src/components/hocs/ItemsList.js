import React  from 'react';

import { Card } from 'semantic-ui-react';

//TODO: clerify id everywhere 
function ItemsList({data, Component}) {
  return (
    <Card.Group>
      {
        data.map(element => 
          <Component 
            key={element.id} 
            item={element} 
          />) 
      }
    </Card.Group>
  );
}

export { ItemsList };
