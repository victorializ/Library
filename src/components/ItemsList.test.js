import React from 'react';
import { shallow } from 'enzyme';

import { ItemsList } from './ItemsList';

describe('ItemsList component', () => {
    it('renders correctly', () => {
        const data = [{id: 1, item: {number: 1}}];
        const Component = item => <p>{item.number}</p>;
        expect(shallow(<ItemsList data={data} Component={Component} />)).toMatchSnapshot();
    });   
});