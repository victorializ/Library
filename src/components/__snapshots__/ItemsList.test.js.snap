// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList component renders correctly 1`] = `
<CardGroup>
  <Component
    item={
      Object {
        "id": 1,
        "item": Object {
          "number": 1,
        },
      }
    }
    key="1"
  />
</CardGroup>
`;
